apps/backend/src/mutations/TechniqueMutations.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  assignInstrumentsToTechniqueValidationSchema,
+  createTechniqueValidationSchema,
+  deleteTechniqueValidationSchema,
+  removeInstrumentsFromTechniqueValidationSchema,
+  updateTechniqueValidationSchema,
+} from '@user-office-software/duo-validation';
+import { inject, injectable } from 'tsyringe';
+
+import { UserAuthorization } from '../auth/UserAuthorization';
+import { Tokens } from '../config/Tokens';
+import { Authorized, EventBus, ValidateArgs } from '../decorators';
+import { Event } from '../events/event.enum';
+import { Rejection, rejection } from '../models/Rejection';
+import { Roles } from '../models/Role';
+import { Technique } from '../models/Technique';
+import { UserWithRole } from '../models/User';
+import { AssignInstrumentsToTechniqueArgs } from '../resolvers/mutations/AssignInstrumentsToTechnique';
+import { CreateTechniqueArgs } from '../resolvers/mutations/CreateTechniqueMutation';
+import { RemoveInstrumentFromTechniqueArgs } from '../resolvers/mutations/RemoveInstrumentFromTechnique';
+import { UpdateTechniqueArgs } from '../resolvers/mutations/UpdateTechniqueMutations';
+import { TechniqueDataSource } from './../datasources/TechniqueDataSource';
+
+@injectable()
+export default class TechniqueMutations {
+  constructor(
+    @inject(Tokens.TechniqueDataSource)
+    private dataSource: TechniqueDataSource,
+    @inject(Tokens.UserAuthorization) private userAuth: UserAuthorization
+  ) {}
+
+  @EventBus(Event.TECHNIQUE_CREATED)
+  @ValidateArgs(createTechniqueValidationSchema)
```

Comment:
I just went through the schema in the user-office-lib. I felt it could be improved a little bit. 

https://github.com/UserOfficeProject/user-office-lib/blob/master/packages/validation/src/Technique/index.ts#L5

For ex., The Short Code in DB has the length restriction of 20 as defined in the migration. However in the validation, no such condition is present. I think it is a good idea to align the DB level validation, backend Level validation and frontend level on par with each other.

Reply:
Added size restrictions for name and short code. Created a new PR https://github.com/UserOfficeProject/user-office-lib/pull/195